### PR TITLE
fix(iOS): fix rendering `RCTRedBoxExtraData`

### PR DESCRIPTION
**Pull request by krozniata**
## Summary:

This PR fixes rendering of `RCTRedBoxExtraData`

I noticed that it wasn't displaying the `reload` and `dismiss` buttons, which made it impossible to close modal and to reload JS on e.g. `visionOS` (on `iOS` it could only be closed by swiping).

PR adds these buttons back and also introduces some refactoring

Before & After:

pr-img

## Changelog:

[IOS] [FIXED] - Fix rendering `RCTRedBoxExtraData`

## Test Plan:

Make sure that `RCTRedBoxExtraData`  displays and works as expected

### CODE REVIEW

File: packages/react-native/React/Modules/RCTRedBoxExtraDataViewController.m [hunk 1]
```diff
@@ -71,72 +73,82 @@ - (instancetype)init
...
+  [button setTitle:title forState:UIControlStateNormal];
+  [button setTitleColor:[UIColor colorWithWhite:1 alpha:1] forState:UIControlStateNormal];
+  [button setTitleColor:[UIColor colorWithWhite:1 alpha:0.5] forState:UIControlStateHighlighted];
+  
+  [button addTarget:self action:action forControlEvents:UIControlEventTouchUpInside];
+  return button;
+}
 
+-(void)viewDidLoad {
+  CGFloat bottomSafeAreaInset = RCTSharedApplication().delegate.window.safeAreaInsets.bottom;
```
Comment (okwasniewski) [inline packages/react-native/React/Modules/RCTRedBoxExtraDataViewController.m:97]: I think it's better to use `RCTKeyWindow()` to retrieve the main one

File: packages/react-native/React/Modules/RCTRedBoxExtraDataViewController.m [hunk 2]
```diff
@@ -71,72 +73,82 @@ - (instancetype)init
...
+  
+  CGFloat buttonHeight = 60;
+  CGFloat buttonWidth = self.view.bounds.size.width / 2;
+  
+  CGRect frame = self.view.bounds;
+  frame.size.height -= buttonHeight + bottomSafeAreaInset;
+  
+  _tableView = [[UITableView alloc] initWithFrame:frame style:UITableViewStylePlain];
+  _tableView.autoresizingMask = UIViewAutoresizingFlexibleWidth | UIViewAutoresizingFlexibleHeight;
```
Comment (okwasniewski) [inline packages/react-native/React/Modules/RCTRedBoxExtraDataViewController.m:106]: How about making this table view also use auto layout?

Comment (krozniata) [reply packages/react-native/React/Modules/RCTRedBoxExtraDataViewController.m]: Agree!

**Comment by lunaleaps** [comment]
Could you clarify what broke and how this fixes it?

**Comment by krozniata** [comment]
@lunaleaps 
Sure! Before, when user opened the "Extra data" section in `RedBox` it was impossible to reload JS from it, because due to a bug in the layout, buttons were not visible and the modal could only be dismissed by dragging it down (which wasn't possible to do in `visionOS`, so it couldn't be dismissed at all). 

This PR brings back the missing buttons by fixing the `RCTRedBoxExtraData` layout, as they were already added before they just weren't visible.

**Comment by krozniata** [comment]
@lunaleaps 
Is there anything else that's needs to be done to merge this PR?

**Comment by Biki-das** [comment]
@krozniata could you show me when this appears with an example? like the bug

**Comment by krozniata** [comment]
@cipolleschi 👀

**Review by cipolleschi — APPROVED** [review]
Thank you so much for this changes and for your patience.
This PR slipped from me until today.

**Pull request closed without merging**